### PR TITLE
Adds ev charging measure to example project

### LINKED ISSUE #183: Add EV charging measure to example project

### PR DESCRIPTION
**Pull request by tanushree04**
### Pull Request Description

Adds measure for ev-carging load and curtailment to baseworkflow, creates and ev charging mapper and adds it to the example files to create an example project using the cli.
This will allow users to add ev-charging to features from their feature file by specifying whether ev-charging is true for that feature and adding related fields in the feature file.

### Checklist (Delete lines that don't apply)

- [x] Unit tests have been added or updated
- [x] All ci tests pass (green)
- [x] An [issue](https://github.com/urbanopt/urbanopt-cli/issues) has been created (which will be used for the changelog)
- [x] This branch is up-to-date with develop

### CODE REVIEW

File: example_files/example_project.json [hunk 1]
```diff
@@ -118,7 +118,11 @@
         "building_type": "Food service",
         "floor_area": 22313,
         "footprint_area": 22313,
-        "number_of_stories": 1
+        "number_of_stories": 1,
+        "ev_charging": true,
```
Comment (tanushree04) [inline example_files/example_project.json:122]: `ev_charging` needs to be set to true when a user wants to add ev_charging to the feature. 
When set to true, users can provide additional arguments for the ev_charging measure or rely on the defaults used in the measure.

File: example_files/mappers/EvCharging.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          elsif typical_work.include? building_type
+            return 'Typical Work'
+          end
+        end
+  
+        # add EV loads
+        ev_charging = nil
+
+        begin
+          if feature.ev_charging == true
```
Comment (tanushree04) [inline example_files/mappers/EvCharging.rb:66]: If ev_charging is true, the `add_ev_load` measure is added to the feature.

File: example_files/mappers/EvCharging.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        end
+
+        if ev_charging == true
+          begin
+            ev_charging_station_type = feature.ev_charging_station_type
+          rescue
+          end
+          if ev_charging_station_type.nil? or ev_charging_station_type.empty?
+            building_type = feature.building_type
+            ev_charging_station_type = ev_charging_type(building_type)
```
Comment (tanushree04) [inline example_files/mappers/EvCharging.rb:80]: The ev_charging_station_type is inferred from the building type it is not specified in the feature file.

**Review by kflemin — APPROVED** [review]

File: Gemfile [hunk 4]
```diff
@@ -60,5 +60,5 @@ allow_local = ENV['FAVOR_LOCAL_GEMS']
 # if allow_local && File.exist?('../urbanopt-reporting-gem')
   # gem 'urbanopt-reporting', path: '../urbanopt-reporting-gem'
 # elsif allow_local
-  gem 'urbanopt-reporting', github: 'URBANopt/urbanopt-reporting-gem', branch: 'develop'
+  gem 'urbanopt-reporting', github: 'URBANopt/urbanopt-reporting-gem', branch: 'add_EV_results'
```
Comment (vtnate) [inline Gemfile:63]: Need to make this same change in `example_files/Gemfile` for tests to pass.

Comment (tanushree04) [reply Gemfile:63]: @vtnate still working on this PR

File: example_files/visualization/input_visualization_feature.html [hunk 5]
```diff
@@ -384,7 +384,7 @@ <h2>Annual End Uses</h2>
         
         var datasets = ['Electricity:Facility', 'ElectricityProduced:Facility', 'Gas:Facility'];
 
-        var endUseKeys = ['Heating:Electricity', 'Cooling:Electricity', 'InteriorLights:Electricity', 'ExteriorLights:Electricity', 'InteriorEquipment:Electricity', 'Fans:Electricity', 'Pumps:Electricity', 'HeatRejection:Electricity', 'WaterSystems:Electricity'];
+        var endUseKeys = ['Heating:Electricity', 'Cooling:Electricity', 'InteriorLights:Electricity', 'ExteriorLights:Electricity', 'ExteriorEquipment:Electricity', 'InteriorEquipment:Electricity', 'Fans:Electricity', 'Pumps:Electricity', 'HeatRejection:Electricity', 'WaterSystems:Electricity'];
```
Comment (tanushree04) [inline example_files/visualization/input_visualization_feature.html:387]: ExteriorEquipment:Electricity is added to feature reports

File: example_files/visualization/input_visualization_scenario.html [hunk 6]
```diff
@@ -375,7 +375,7 @@ <h2>Annual End Uses</h2>
         
         var datasets = ['Electricity:Facility', 'ElectricityProduced:Facility', 'Gas:Facility'];
 
-        var endUseKeys = ['Heating:Electricity', 'Cooling:Electricity', 'InteriorLights:Electricity', 'ExteriorLights:Electricity', 'InteriorEquipment:Electricity', 'Fans:Electricity', 'Pumps:Electricity', 'HeatRejection:Electricity', 'WaterSystems:Electricity'];
+        var endUseKeys = ['Heating:Electricity', 'Cooling:Electricity', 'InteriorLights:Electricity', 'ExteriorLights:Electricity', 'InteriorEquipment:Electricity', 'ExteriorEquipment:Electricity', 'Fans:Electricity', 'Pumps:Electricity', 'HeatRejection:Electricity', 'WaterSystems:Electricity'];
```
Comment (tanushree04) [inline example_files/visualization/input_visualization_scenario.html:378]: ExteriorEquipment:Electricity is added to scenario reports

**Comment by vtnate** [comment]
@tanushree04 What are the steps I need to take to see the EV results? I did a run, default process, and visualize of buildings 2 & 5, and didn't see any EV results in building 2. Do I need to adjust the feature file?

**Comment by tanushree04** [comment]
@vtnate The results are in the ExteriorEquipment:Electricity for the Feature 2 in the Feaure reports and show up in Annual End uses graph in the visualization.

**Comment by vtnate** [comment]
When I run `visualize` I get this output, which makes me think I did something wrong:

Screen Shot 2021-02-01 at 9 11 54 AM

**Comment by tanushree04** [comment]
@vtnate once you create the scenario files, are you running the evcharging_scenario.csv?

**Comment by vtnate** [comment]
> @vtnate once you create the scenario files, are you running the evcharging_scenario.csv?

Yes.

**Review by kflemin — APPROVED** [review]
👍

**Review by vtnate — APPROVED** [review]
I had an unrelated error that was very sneakily preventing this from working. Once I was finally able to test it as intended, everything worked as intended. :shipit:

**Pull request merged**